CMSIS/NN/Include/arm_nnsupportfunctions.h

```diff
@@ -54,6 +54,12 @@ extern "C" {
     ((((int32_t)(v0) << 0) & (int32_t)0x000000FF) | (((int32_t)(v1) << 8) & (int32_t)0x0000FF00) |                     \
      (((int32_t)(v2) << 16) & (int32_t)0x00FF0000) | (((int32_t)(v3) << 24) & (int32_t)0xFF000000))
 
+#if defined(__ARMCC_VERSION)
```

Comment:
__RESTRICT is defined in cmsis_gcc/armclang.h files.. Let's use that instead. And are we sure to use restrict only for armclang?